samples/nrf9160/nrf_cloud_mqtt_multi_service/src/application.c

```diff
@@ -0,0 +1,259 @@
+			log_strdup(sensor));
+		goto cleanup;
+	}
+
+	/* Send the sensor sample container object as a device message. */
+	ret = send_device_message_cJSON(msg_obj);
+
+cleanup:
+	if (msg_obj) {
+		cJSON_Delete(msg_obj);
+	}
+	return ret;
+}
+
+/**
+ * @brief Transmit a collected GNSS sample to nRF Cloud.
+ *
+ * @param NMEA - The raw MNEA value to send to nRF Cloud.
+ * @return int - 0 on success, negative error code otherwise.
+ */
+static int send_gnss_nmea(const char *const nmea)
+{
+	int ret = 0;
+
+	/* Create a timestamped message container object for the GNSS NMEA data. */
+	cJSON *msg_obj = create_timestamped_data_message_object("GPS");
+
+	if (!msg_obj) {
+		ret = -EINVAL;
+		goto cleanup;
+	}
+
+	/* Populate the container object with the NMEA data. */
+	if (cJSON_AddStringToObject(msg_obj, NRF_CLOUD_JSON_DATA_KEY, nmea) == NULL) {
+		ret = -ENOMEM;
+		LOG_ERR("Failed to append NMEA data to GNSS message object");
+		goto cleanup;
+	}
+
+	/* Send the sensor sample container object as a device message. */
+	ret = send_device_message_cJSON(msg_obj);
+
+cleanup:
+	if (msg_obj) {
+		cJSON_Delete(msg_obj);
+	}
+	return ret;
+}
+
+
+/**
+ * @brief Compute checksum for NMEA string.
+ *
+ * @param datastring - The NMEA string to compute the checksum for.
+ * @return uint8_t - The resultant checksum.
+ */
+static uint8_t nmea_checksum(const char *const datastring)
+{
+	uint8_t checksum = 0;
+	int len = strlen(datastring);
+
+	for (int i = 0; i < len; i++) {
+		checksum ^= datastring[i];
+	}
+	return checksum;
+}
+
+/**
+ * @brief Callback to receive periodic location updates from location_tracking.c and forward them
+ * to nRF Cloud.
+ *
+ * Note that cellular positioning (MCELL/Multi-Cell and SCELL/Single-Cell) is sent to nRF
+ * Cloud automatically (since the Location library and nRF Cloud must work together to
+ * determine them in the first place). GNSS positions, on the other hand, must be
+ * sent manually, since they are determined entirely on-device.
+ *
+ * @param location_data - The received location update.
+ *
+ */
+static void on_location_update(const struct location_data location_data)
+{
+	LOG_INF("Location Updated: %.06f N %.06f W, accuracy: %.01f m, Method: %s",
+		location_data.latitude, location_data.longitude, location_data.accuracy,
+		location_data.method == LOCATION_METHOD_CELLULAR	? "Cellular" :
+		location_data.method == LOCATION_METHOD_GNSS		? "GNSS"     :
+		location_data.method == LOCATION_METHOD_WIFI		? "WIFI"     :
+									  "Invalid");
+
+	/* If the position update was derived using GNSS, send it onward to nRF Cloud. */
+	if (location_data.method == LOCATION_METHOD_GNSS) {
+		LOG_INF("GNSS Position Update! Sending to nRF Cloud...");
+
+		/* Synthesize an NMEA message from the provided lat/long/timestamp.
+		 * This is necessary, because the nRF Cloud MQTT API only supports NMEA messages.
+		 */
+
+		char nmea_buf[50];
+		int latdeg = floor(fabs(location_data.latitude));
```

Comment:
I agree -- on the fence about appid, though, since `NRF_CLOUD_JSON_APPID_VAL_TEMP` and the like treat it as one word, and the underlying keyname is one word too